core/securitykey.py

```diff
@@ -42,25 +45,33 @@ def create(duration: Union[None, int] = None, **kwargs) -> str:
     if not duration:
         assert not kwargs, "kwargs are not allowed when session security key is wanted"
         return current.session.get().getSecurityKey()
-    key = generateRandomString()
+
+    key = utils.generateRandomString()
     duration = int(duration)
-    dbObj = db.Entity(db.Key(securityKeyKindName, key))
+
+    entity = db.Entity(db.Key(SECURITYKEY_KINDNAME, key))
+
     for k, v in kwargs.items():
-        dbObj[k] = v
-    dbObj["until"] = utcNow() + timedelta(seconds=duration)
-    db.Put(dbObj)
+        entity[k] = v
```

Comment:
Just call `.update()` or use the merge operator:
```suggestion
   entity |= kwargs
```